lighthouse-core/gather/driver.js

```diff
@@ -347,20 +346,19 @@ class Driver {
   }
 
   /**
-   * If our main document URL redirects, we will update options.url accordingly
-   * As such, options.url will always represent the post-redirected URL.
-   * options.initialUrl is the pre-redirect URL that things started with
-   * @param {!Object} opts
+   * Track any redirects of the supplied url. options.url will always represent
+   * the post-redirected URL.
+   * @param {{url: string}} options
    */
-  enableUrlUpdateIfRedirected(opts) {
-    this._networkRecorder.on('requestloaded', redirectRequest => {
+  enableUrlUpdateIfRedirected(options) {
```

Comment:
let's rename away from options now that it's not mutating the actual options object